backends/firrtl/firrtl.cc

```diff
@@ -392,7 +392,37 @@ struct FirrtlWorker
 		return result;
 	}
 
-	void run()
+	void emit_extmodule()
+	{
+		std::string moduleFileinfo = getFileinfo(module);
+		f << stringf("  extmodule %s: %s\n", make_id(module->name), moduleFileinfo.c_str());
+		vector<string> port_decls;
+
+		for (auto wire : module->wires())
+		{
+			const auto wireName = make_id(wire->name);
+			std::string wireFileinfo = getFileinfo(wire);
+
+			// Maybe not needed?
+			if (wire->port_id)
+			{
+				if (wire->port_input && wire->port_output)
+				{
+					log_error("Module port %s.%s is inout!\n", log_id(module), log_id(wire));
+				}
+				port_decls.push_back(stringf("    %s %s: UInt<%d> %s\n", wire->port_input ? "input" : "output",
+						wireName, wire->width, wireFileinfo.c_str()));
+			}
+		}
+
+		for (auto str : port_decls) {
+			f << str;
+		}
+
+		f << stringf("\n");
```

Comment:
You don't need `stringf()` if you're not using the format specifier.

```suggestion
		f << "\n";
```
(and on line 1110)

Reply:
I agree that there is no format specifier here and calling this function doesn't help, but the rest of the code used `stringf` everywhere as a convention and I stuck to it. I can go ahead and change it everywhere if you think it would be necessary?

Reply:
I don't think it's necessary, no.  Frankly I don't think it makes much of a difference; I was just pointing it out.